### ISSUE TITLE
Option to not raise request to keep screen awake when streaming video

### ISSUE THREAD

**Issue by LewisSpring** [SoftFever/OrcaSlicer#7738]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

When streaming video from the printer, OrcaSlicer raises the following powercfg /requests  flags, keeping the screen on even when the user may not be at the computer

```
powercfg /requests
DISPLAY:
[PROCESS] \Device\HarddiskVolume1\Program Files\OrcaSlicer\orca-slicer.exe

SYSTEM:
[PROCESS] \Device\HarddiskVolume1\Program Files\OrcaSlicer\orca-slicer.exe

AWAYMODE:
None.

EXECUTION:
None.

PERFBOOST:
None.

ACTIVELOCKSCREEN:
None.
```

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

A setting to disable OrcaSlicer from raising any "stay awake" flags while streaming video, allowing the computer to lock and the screen to standby even while the video plays.

There could even be an option to stop streaming when the computer locks to save Printer/Computer resources.

if OrcaSlicer has been left streaming in the background, it also keeps raising the flag, so if you are forgetful it is not immediately obvious that OrcaSlicer is prohibiting lock/sleep.

### Describe alternatives you've considered

1. Remember to stop video streaming

**Comment by Catriks** [comment]
I have this issue as well. The power request, that prevents screens from turning off, is activated when I enable camera stream from P1S, but the request does not go away even if the stream is stopped. It only seems to go away after shutting down Orca completely. 

I am using 2.3.1-dev portable.

Cany anyone confirm is this power request is intentional, but mis-behaving, or fully unintentional?

**Comment by jm4n69** [comment]
Here in February 2026, Orca 2.3.1, this is still an issue for me. Specifically on the P1S (for example, with the Elegoo CC1 since it's a web interface, stopping video playback there, allows the PC go to sleep).

With the P1S, regardless of whether I am streaming video or which "tab" I'm in (or if Orca is minimized, etc), I get similar output to the OP:

```
C:\Windows\System32>powercfg /requests
DISPLAY:
[PROCESS] \Device\HarddiskVolume3\Program Files\OrcaSlicer\orca-slicer.exe

SYSTEM:
[PROCESS] \Device\HarddiskVolume3\Program Files\OrcaSlicer\orca-slicer.exe

AWAYMODE:
None.

EXECUTION:
None.

PERFBOOST:
None.

ACTIVELOCKSCREEN:
None.
```

That's with the P1S finished printing and Orca just sitting on the "prepare" tab.

**Comment by Catriks** [comment]
It seems that since 2.3.1-dev, when I last commented here, it has been fixed that Orca no longer blocks sleep _after_ video streaming is stopped. Before that, it would keep being blocked until Orca was closed completely. So thank you for fixing that.

But during video playback, sleep is still blocked on 2.3.1 stable and 2.3.2 beta. I still think this should be a toggleable option as suggested by OP - and preferrably set to not block it by default.

My argument is that it is not common that monitoring a printer is something someone would do for long periods of time without doing anything else on the computer (therefore not comparable to video players, that also block sleep). Monitoring the printer is also not useful if there is no one at the computer. Therefore sleep, and locking of the user account, being blocked by the video stream is unexpected and unwanted behaviour in my opinion.

**Comment by jm4n69** [comment]
Agreed with Catriks here, but I came back to point out that today, just playing around with powercfg, it seems to be intermittent.

I switch between the (crappy) P1S built-in camera and an external URL (webcam jammed in the other corner lol), and both cases clicking "stop" still showed the same thing in powercfg.

However after restarting Orca (new project later in the day) I can't reproduce this behavior. Clicking play, whichever camera is selected, causes powercfg to show that the screen is held on; clicking Stop removes that entry, again regardless which camera I am using.

I think it starts acting weird when I keep using the same instance of Orca to start a print, then load up another project and switch printers.... which I'd been doing all morning. After a fresh start that behavior quit and it feels "normal". Guess I should just close it out and re-launch between sending a print...

(perhaps I'm not "stopping" the stream before switching machines?)

But as I said I do agree, it would be nice to be able to toggle this ever keeping the PC awake at all. Often, I click Print, and walk away; Orca proceeds to send the file to the printer, then switches to the device tab and starts streaming the player... I come back an hour or more later to find my screen awake...
